hclsyntax/parse_traversal_test.go

```diff
@@ -208,10 +210,61 @@ func TestParseTraversalAbs(t *testing.T) {
 			},
 			1, // extra junk after traversal
 		},
+
+		{
+			"foo[*]",
+			hcl.Traversal{
+				hcl.TraverseRoot{
+					Name: "foo",
+					SrcRange: hcl.Range{
+						Start: hcl.Pos{Line: 1, Column: 1, Byte: 0},
+						End:   hcl.Pos{Line: 1, Column: 4, Byte: 3},
+					},
+				},
+				hcl.TraverseSplat{
+					SrcRange: hcl.Range{
+						Start: hcl.Pos{Line: 1, Column: 4, Byte: 3},
+						End:   hcl.Pos{Line: 1, Column: 7, Byte: 6},
+					},
+				},
+			},
+			0,
+		},
+		{
+			"foo.*", // Still not supporting this.
+			hcl.Traversal{
+				hcl.TraverseRoot{
+					Name: "foo",
+					SrcRange: hcl.Range{
+						Start: hcl.Pos{Line: 1, Column: 1, Byte: 0},
+						End:   hcl.Pos{Line: 1, Column: 4, Byte: 3},
+					},
+				},
+			},
+			1,
+		},
+		{
+			"foo[*].bar", // Run this through the unsupported function.
+			hcl.Traversal{
+				hcl.TraverseRoot{
+					Name: "foo",
+					SrcRange: hcl.Range{
+						Start: hcl.Pos{Line: 1, Column: 1, Byte: 0},
+						End:   hcl.Pos{Line: 1, Column: 4, Byte: 3},
+					},
+				},
+			},
+			1,
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.src, func(t *testing.T) {
+			if test.src == "foo[*]" {
+				// Skip the test that introduces splat syntax.
+				t.Skip("skipping test for unsupported splat syntax")
```

Comment:
Do I understand this right that we skip it here since we use `ParseTraversalAbs` in which case this is unsupported, but include it in the partial_ test cases since `ParseTraversalPartial` supports it?

Reply:
Yeah, exactly. Basically, I wanted to make sure that using the splat was still broken in the old implementation but worked in the new one. I figured the easiest way to do that was to just manually skip the relevant tests since there's only one of each branch that needs to be skipped. 

I could complicate the whole function by adding in some additional metadata to the test cases, but thought it was overkill just for one case.